### ISSUE TITLE
KafkaClient.close() doesn't fully clean up

### ISSUE THREAD

**Issue by twm** [ciena/afkak#71]
`KakfaClient.close()` terminates outstanding connections and fails operations that were waiting on requests in those brokers, but the client may internally retry due to those failures. This can lead to both:

* Connection attempts in the endpoint_factory
* Delayed calls in the reactor (for timeouts)

The reactor should be clean once the deferred returned by `close()` has fired.

This is of less impact than it might seem because Producer and Consumer clean up outstanding requests when closed. Once that happens `KafkaClient.close()` is sufficient.
